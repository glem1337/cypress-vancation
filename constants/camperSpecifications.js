import yup from 'lib/yupLocalised';
import {
  FINISH_RANGE_LENGTH_CAMPER,
  MAX_SEAT_SLEEP_PLACE,
  MIN_SEAT_SLEEP_PLACE,
  START_RANGE_LENGTH_CAMPER,
} from './camper';

export const SPECIFICATIONS_VALIDATION_SCHEMA = {
  vehicleTypeName: yup.string().required({
    id: 'validations.cantBeBlank',
    values: { pref: 'Vehicle type' },
  }),
  name: yup
    .string()
    .required({ id: 'validations.cantBeBlank', values: { pref: 'Make' } }),
  modelNaming: yup
    .string()
    .required({ id: 'validations.cantBeBlank', values: { pref: 'Model' } }),
  year: yup
    .string()
    .required({ id: 'validations.cantBeBlank', values: { pref: 'Year' } }),
  length: yup
    .number()
    .required({
      id: 'validations.cantBeBlank',
      values: { pref: 'Length' },
    })
    .min(START_RANGE_LENGTH_CAMPER, { id: 'addNewCamper.specifications.validations.length.min', values: { value: START_RANGE_LENGTH_CAMPER } })
    .max(FINISH_RANGE_LENGTH_CAMPER, { id: 'addNewCamper.specifications.validations.length.max', values: { value: FINISH_RANGE_LENGTH_CAMPER } }),
  insideHeight: yup.string().required({
    id: 'validations.cantBeBlank',
    values: { pref: 'Inside height' },
  }),
  seats: yup
    .number()
    .required({ id: 'validations.cantBeBlank', values: { pref: 'Seats' } })
    .min(MIN_SEAT_SLEEP_PLACE, { id: 'addNewCamper.specifications.validations.seats.min', values: { value: MIN_SEAT_SLEEP_PLACE } })
    .max(MAX_SEAT_SLEEP_PLACE, { id: 'addNewCamper.specifications.validations.seats.max', values: { value: MAX_SEAT_SLEEP_PLACE } }),
  sleeps: yup
    .number()
    .required({ id: 'validations.cantBeBlank', values: { pref: 'Sleeps' } })
    .min(MIN_SEAT_SLEEP_PLACE, { id: 'addNewCamper.specifications.validations.sleeps.min', values: { value: MIN_SEAT_SLEEP_PLACE } })
    .max(MAX_SEAT_SLEEP_PLACE, { id: 'addNewCamper.specifications.validations.sleeps.max', values: { value: MAX_SEAT_SLEEP_PLACE } }),
  camperLocation: yup.object().geocoder({
    id: 'validations.cantBeBlank',
    values: { pref: 'Camper location' },
  }),
  stateRegistred: yup.string().required({
    id: 'validations.cantBeBlank',
    values: { pref: 'State registered' },
  }),
  transmission: yup.string().required({
    id: 'validations.cantBeBlank',
    values: { pref: 'Transmission' },
  }),
  mileage: yup.string().required({
    id: 'validations.cantBeBlank',
    values: { pref: 'Current mileage (estimate)' },
  }),
  fuelType: yup.string().required({
    id: 'validations.cantBeBlank',
    values: { pref: 'Fuel type' },
  }),
  whoBuiltCamper: yup.string(),
  drivetrain: yup.string().required({
    id: 'validations.cantBeBlank',
    values: { pref: 'Drivetrain' },
  }),
};

export const WATER_FORM_VALIDATION = {
  MIN: 0,
  MAX: 60,
};
