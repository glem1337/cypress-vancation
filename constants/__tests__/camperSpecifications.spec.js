import * as headers from '../camperSpecifications';

it('camperSpecifications constants matches snapshot', () => {
  expect(headers).toMatchSnapshot();
});
