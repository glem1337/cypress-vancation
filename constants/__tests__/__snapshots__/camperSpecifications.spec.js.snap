// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`camperSpecifications constants matches snapshot 1`] = `
Object {
  "SPECIFICATIONS_VALIDATION_SCHEMA": Object {
    "camperLocation": ObjectSchema {
      "_blacklist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_blacklistError": undefined,
      "_excludedEdges": Array [],
      "_nodes": Array [],
      "_sortErrors": [Function],
      "_typeError": [Function],
      "_whitelist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_whitelistError": undefined,
      "conditions": Array [],
      "deps": Array [],
      "exclusiveTests": Object {
        "geocoder": false,
      },
      "fields": Object {},
      "spec": Object {
        "abortEarly": true,
        "nullable": false,
        "presence": "optional",
        "recursive": true,
        "strict": false,
        "strip": false,
      },
      "tests": Array [
        [Function],
      ],
      "transforms": Array [
        [Function],
      ],
      "type": "object",
    },
    "drivetrain": StringSchema {
      "_blacklist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_blacklistError": undefined,
      "_mutate": undefined,
      "_typeError": [Function],
      "_whitelist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_whitelistError": undefined,
      "conditions": Array [],
      "deps": Array [],
      "exclusiveTests": Object {
        "required": true,
      },
      "spec": Object {
        "abortEarly": true,
        "nullable": false,
        "presence": "required",
        "recursive": true,
        "strict": false,
        "strip": false,
      },
      "tests": Array [
        [Function],
      ],
      "transforms": Array [
        [Function],
      ],
      "type": "string",
    },
    "fuelType": StringSchema {
      "_blacklist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_blacklistError": undefined,
      "_mutate": undefined,
      "_typeError": [Function],
      "_whitelist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_whitelistError": undefined,
      "conditions": Array [],
      "deps": Array [],
      "exclusiveTests": Object {
        "required": true,
      },
      "spec": Object {
        "abortEarly": true,
        "nullable": false,
        "presence": "required",
        "recursive": true,
        "strict": false,
        "strip": false,
      },
      "tests": Array [
        [Function],
      ],
      "transforms": Array [
        [Function],
      ],
      "type": "string",
    },
    "insideHeight": StringSchema {
      "_blacklist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_blacklistError": undefined,
      "_mutate": undefined,
      "_typeError": [Function],
      "_whitelist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_whitelistError": undefined,
      "conditions": Array [],
      "deps": Array [],
      "exclusiveTests": Object {
        "required": true,
      },
      "spec": Object {
        "abortEarly": true,
        "nullable": false,
        "presence": "required",
        "recursive": true,
        "strict": false,
        "strip": false,
      },
      "tests": Array [
        [Function],
      ],
      "transforms": Array [
        [Function],
      ],
      "type": "string",
    },
    "length": NumberSchema {
      "_blacklist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_blacklistError": undefined,
      "_typeError": [Function],
      "_whitelist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_whitelistError": undefined,
      "conditions": Array [],
      "deps": Array [],
      "exclusiveTests": Object {
        "max": true,
        "min": true,
        "required": true,
      },
      "spec": Object {
        "abortEarly": true,
        "nullable": false,
        "presence": "required",
        "recursive": true,
        "strict": false,
        "strip": false,
      },
      "tests": Array [
        [Function],
        [Function],
        [Function],
      ],
      "transforms": Array [
        [Function],
      ],
      "type": "number",
    },
    "mileage": StringSchema {
      "_blacklist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_blacklistError": undefined,
      "_mutate": undefined,
      "_typeError": [Function],
      "_whitelist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_whitelistError": undefined,
      "conditions": Array [],
      "deps": Array [],
      "exclusiveTests": Object {
        "required": true,
      },
      "spec": Object {
        "abortEarly": true,
        "nullable": false,
        "presence": "required",
        "recursive": true,
        "strict": false,
        "strip": false,
      },
      "tests": Array [
        [Function],
      ],
      "transforms": Array [
        [Function],
      ],
      "type": "string",
    },
    "modelNaming": StringSchema {
      "_blacklist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_blacklistError": undefined,
      "_mutate": undefined,
      "_typeError": [Function],
      "_whitelist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_whitelistError": undefined,
      "conditions": Array [],
      "deps": Array [],
      "exclusiveTests": Object {
        "required": true,
      },
      "spec": Object {
        "abortEarly": true,
        "nullable": false,
        "presence": "required",
        "recursive": true,
        "strict": false,
        "strip": false,
      },
      "tests": Array [
        [Function],
      ],
      "transforms": Array [
        [Function],
      ],
      "type": "string",
    },
    "name": StringSchema {
      "_blacklist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_blacklistError": undefined,
      "_mutate": undefined,
      "_typeError": [Function],
      "_whitelist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_whitelistError": undefined,
      "conditions": Array [],
      "deps": Array [],
      "exclusiveTests": Object {
        "required": true,
      },
      "spec": Object {
        "abortEarly": true,
        "nullable": false,
        "presence": "required",
        "recursive": true,
        "strict": false,
        "strip": false,
      },
      "tests": Array [
        [Function],
      ],
      "transforms": Array [
        [Function],
      ],
      "type": "string",
    },
    "seats": NumberSchema {
      "_blacklist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_blacklistError": undefined,
      "_typeError": [Function],
      "_whitelist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_whitelistError": undefined,
      "conditions": Array [],
      "deps": Array [],
      "exclusiveTests": Object {
        "max": true,
        "min": true,
        "required": true,
      },
      "spec": Object {
        "abortEarly": true,
        "nullable": false,
        "presence": "required",
        "recursive": true,
        "strict": false,
        "strip": false,
      },
      "tests": Array [
        [Function],
        [Function],
        [Function],
      ],
      "transforms": Array [
        [Function],
      ],
      "type": "number",
    },
    "sleeps": NumberSchema {
      "_blacklist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_blacklistError": undefined,
      "_typeError": [Function],
      "_whitelist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_whitelistError": undefined,
      "conditions": Array [],
      "deps": Array [],
      "exclusiveTests": Object {
        "max": true,
        "min": true,
        "required": true,
      },
      "spec": Object {
        "abortEarly": true,
        "nullable": false,
        "presence": "required",
        "recursive": true,
        "strict": false,
        "strip": false,
      },
      "tests": Array [
        [Function],
        [Function],
        [Function],
      ],
      "transforms": Array [
        [Function],
      ],
      "type": "number",
    },
    "stateRegistred": StringSchema {
      "_blacklist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_blacklistError": undefined,
      "_mutate": undefined,
      "_typeError": [Function],
      "_whitelist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_whitelistError": undefined,
      "conditions": Array [],
      "deps": Array [],
      "exclusiveTests": Object {
        "required": true,
      },
      "spec": Object {
        "abortEarly": true,
        "nullable": false,
        "presence": "required",
        "recursive": true,
        "strict": false,
        "strip": false,
      },
      "tests": Array [
        [Function],
      ],
      "transforms": Array [
        [Function],
      ],
      "type": "string",
    },
    "transmission": StringSchema {
      "_blacklist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_blacklistError": undefined,
      "_mutate": undefined,
      "_typeError": [Function],
      "_whitelist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_whitelistError": undefined,
      "conditions": Array [],
      "deps": Array [],
      "exclusiveTests": Object {
        "required": true,
      },
      "spec": Object {
        "abortEarly": true,
        "nullable": false,
        "presence": "required",
        "recursive": true,
        "strict": false,
        "strip": false,
      },
      "tests": Array [
        [Function],
      ],
      "transforms": Array [
        [Function],
      ],
      "type": "string",
    },
    "vehicleTypeName": StringSchema {
      "_blacklist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_blacklistError": undefined,
      "_mutate": undefined,
      "_typeError": [Function],
      "_whitelist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_whitelistError": undefined,
      "conditions": Array [],
      "deps": Array [],
      "exclusiveTests": Object {
        "required": true,
      },
      "spec": Object {
        "abortEarly": true,
        "nullable": false,
        "presence": "required",
        "recursive": true,
        "strict": false,
        "strip": false,
      },
      "tests": Array [
        [Function],
      ],
      "transforms": Array [
        [Function],
      ],
      "type": "string",
    },
    "whoBuiltCamper": StringSchema {
      "_blacklist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_mutate": undefined,
      "_typeError": [Function],
      "_whitelist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "conditions": Array [],
      "deps": Array [],
      "exclusiveTests": Object {},
      "spec": Object {
        "abortEarly": true,
        "nullable": false,
        "presence": "optional",
        "recursive": true,
        "strict": false,
        "strip": false,
      },
      "tests": Array [],
      "transforms": Array [
        [Function],
      ],
      "type": "string",
    },
    "year": StringSchema {
      "_blacklist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_blacklistError": undefined,
      "_mutate": undefined,
      "_typeError": [Function],
      "_whitelist": ReferenceSet {
        "list": Set {},
        "refs": Map {},
      },
      "_whitelistError": undefined,
      "conditions": Array [],
      "deps": Array [],
      "exclusiveTests": Object {
        "required": true,
      },
      "spec": Object {
        "abortEarly": true,
        "nullable": false,
        "presence": "required",
        "recursive": true,
        "strict": false,
        "strip": false,
      },
      "tests": Array [
        [Function],
      ],
      "transforms": Array [
        [Function],
      ],
      "type": "string",
    },
  },
  "WATER_FORM_VALIDATION": Object {
    "MAX": 60,
    "MIN": 0,
  },
}
`;
